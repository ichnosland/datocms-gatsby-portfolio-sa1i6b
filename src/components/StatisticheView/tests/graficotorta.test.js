import React from 'react';
import { shallow } from 'enzyme';
// import { RadialChart } from 'react-vis';

import GraficoMedieVotiClasse, { FlexibleRadialChart } from '../GraficoTorta';


const mockProps = {
  mediaNazionale: [{
    count: 12,
    title: 'titolo',
    colour: '#fff',
  }, {
    count: 5,
    title: 'titolo 2',
    colour: '#ff0',
  }, {
    count: 0,
    title: 'titolo 2',
    colour: '#00e',
  }],
  onDidascaliaSet: () => { },
  onDidascaliaReset: () => { },
  didascalia: {
    tipologiaGrafico: 'votiClasse',
    display: false,
    tipologia: 'tipologia',
    y: 1,
    x: '#1',
    titolo: 'titolo',
    campioni: 1,
  },
};

describe('<GraficoMedieVotiClasse />', () => {
  it('Mostro la condizione iniziale', () => {
    const renderedComponent = shallow(
      <GraficoMedieVotiClasse
        {...mockProps}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it('Mostro la condizione se hint Ã¨ visibile', () => {
    const props = {
      ...mockProps,
      didascalia: {
        ...mockProps.didascalia,
        display: true,
      },
    };
    const renderedComponent = shallow(
      <GraficoMedieVotiClasse
        {...props}
      />
    );

    expect(renderedComponent).toMatchSnapshot();
  });

  it('setDidascalia deve chiamare onDidascaliaSet', () => {
    const props = {
      ...mockProps,
      onDidascaliaSet: jest.fn(),
    };
    const renderedComponent = shallow(
      <GraficoMedieVotiClasse
        {...props}
      />
    );

    expect(props.onDidascaliaSet).not.toHaveBeenCalled();
    const instance = renderedComponent.instance();
    instance.setDidascalia({
      title: 'titolo',
      tipologia: 'tipologia',
      y: 2,
      totale: 10,
      count: 4,
    });
    expect(props.onDidascaliaSet).toHaveBeenCalledWith({
      display: true,
      tipologiaGrafico: 'votiClasse',
      title: 'titolo',
      titolo: '4 / 10',
      tipologia: 'tipologia',
      y: 2,
      totale: 10,
      count: 4,
      x: '',
    });
  });


  it('VerticalBarSeries(0) onValueMouseOver deve chiamare onDidascaliaSet', () => {
    const props = {
      ...mockProps,
      onDidascaliaSet: jest.fn(),
    };
    const renderedComponent = shallow(
      <GraficoMedieVotiClasse
        {...props}
      />
    );

    const instance = renderedComponent.instance();
    const spy = jest.spyOn(instance, 'setDidascalia');

    renderedComponent.find(FlexibleRadialChart).at(0).props().onValueMouseOver({
      titolo: 'titolo1',
    });
    expect(spy).toHaveBeenCalledWith({ titolo: 'titolo1' });
  });
});
