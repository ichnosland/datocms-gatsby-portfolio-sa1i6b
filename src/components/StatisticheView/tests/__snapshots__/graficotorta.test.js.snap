// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GraficoMedieVotiClasse /> Mostro la condizione iniziale 1`] = `
<StatsElements__ChartBox
  className="ChartBox torta"
>
  <StatsElements__LegendaChartBox>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio 7 -10
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#87deca"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio 5 - 6,99
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#ffe06e"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio 1 - 4,99
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#df2e6d"
      />
    </div>
  </StatsElements__LegendaChartBox>
  <Div
    height="250px"
    margin="0 auto"
    width="250px"
  >
    <FlexibleRadialChart
      animation={true}
      colorDomain={
        Array [
          2,
          1,
          0,
        ]
      }
      colorRange={
        Array [
          "#fff",
          "#ff0",
          "#00e",
        ]
      }
      data={
        Array [
          Object {
            "colour": "#fff",
            "count": 12,
            "title": "titolo",
          },
          Object {
            "colour": "#ff0",
            "count": 5,
            "title": "titolo 2",
          },
          Object {
            "colour": "#00e",
            "count": 0,
            "title": "titolo 2",
          },
        ]
      }
      getAngle={[Function]}
      onValueMouseOut={[Function]}
      onValueMouseOver={[Function]}
    />
  </Div>
</StatsElements__ChartBox>
`;

exports[`<GraficoMedieVotiClasse /> Mostro la condizione se hint Ã¨ visibile 1`] = `
<StatsElements__ChartBox
  className="ChartBox torta"
>
  <StatsElements__LegendaChartBox>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio 7 -10
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#87deca"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio 5 - 6,99
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#ffe06e"
      />
    </div>
    <div>
      <StatsElements__SwatchLabel>
        Punteggio 1 - 4,99
      </StatsElements__SwatchLabel>
      <StatsElements__Swatch
        bgColor="#df2e6d"
      />
    </div>
  </StatsElements__LegendaChartBox>
  <Div
    height="250px"
    margin="0 auto"
    width="250px"
  >
    <FlexibleRadialChart
      animation={true}
      colorDomain={
        Array [
          2,
          1,
          0,
        ]
      }
      colorRange={
        Array [
          "#fff",
          "#ff0",
          "#00e",
        ]
      }
      data={
        Array [
          Object {
            "colour": "#fff",
            "count": 12,
            "title": "titolo",
          },
          Object {
            "colour": "#ff0",
            "count": 5,
            "title": "titolo 2",
          },
          Object {
            "colour": "#00e",
            "count": 0,
            "title": "titolo 2",
          },
        ]
      }
      getAngle={[Function]}
      onValueMouseOut={[Function]}
      onValueMouseOver={[Function]}
    >
      <Hint
        align={
          Object {
            "horizontal": "auto",
            "vertical": "auto",
          }
        }
        format={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.6)",
            "borderRadius": "6px",
            "color": "#fff",
            "fontSize": "14px",
            "padding": "6px 10px",
          }
        }
        value={
          Object {
            "campioni": 1,
            "display": true,
            "tipologia": "tipologia",
            "tipologiaGrafico": "votiClasse",
            "titolo": "titolo",
            "x": "#1",
            "y": 1,
          }
        }
      >
        <div>
          <p>
            tipologia
            : 
            titolo
          </p>
        </div>
      </Hint>
    </FlexibleRadialChart>
  </Div>
</StatsElements__ChartBox>
`;
